src/RemoteInboxNotifications/ProductCountRuleProcessor.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * Rule processor that performs a comparison operation against the number of
+ * products.
+ *
+ * @package WooCommerce Admin/Classes
+ */
+
+namespace Automattic\WooCommerce\Admin\RemoteInboxNotifications;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Rule processor that performs a comparison operation against the number of
+ * products.
+ */
+class ProductCountRuleProcessor implements RuleProcessorInterface {
+	/**
+	 * Constructor.
+	 *
+	 * @param object $product_query The product query.
+	 */
+	public function __construct( $product_query = null ) {
+		$this->product_query = null === $product_query
+			? new \WC_Product_Query(
+				array(
+					'limit'    => 1,
+					'paginate' => true,
+					'return'   => array( 'id' ),
```

Comment:
This should be the string `ids`, not an array.

Reply:
Thanks @jeffstieler this is now fixed